src/Microsoft.AspNetCore.Razor.Language/Syntax/ArrayExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Razor.Language
+{
+    internal static class ArrayExtensions
```

Comment:
Copy paste. Actually, most of the stuff in the `Syntax` folder are not review ready. So can just skim through those. There are a bunch of TODOs such as trivia implementation.

Reply:
Ok, do we need this? I'm pretty gunshy about us having things like `ArrayExtensions` in Razor if we can avoid it.

Reply:
From https://github.com/dotnet/roslyn/blob/d4dab355b96955aca5b4b0ebf6282575fad78ba8/src/Compilers/Core/Portable/InternalUtilities/ArrayExtensions.cs#L186. I'll check if I can get rid of this.